packages/go_router/lib/src/redirection.dart

```diff
@@ -76,7 +77,9 @@ RouteMatchList redirect(RouteMatchList prevMatchList,
     assert(topRoute is GoRoute,
         'Last RouteMatch should contain a GoRoute, but was ${topRoute.runtimeType}');
     final GoRoute topGoRoute = topRoute as GoRoute;
-    final String? topRouteLocation = topGoRoute.redirect(
+    final GoRouterRedirect redirect =
+        topGoRoute.redirect ?? (GoRouterState? state) => null;
```

Comment:
If it is null, why not just bail out the redirect?

Reply:
I was matching the previous behavior but I can clean that up